generator.ts

```diff
@@ -0,0 +1,254 @@
+import { promises as fsPromises } from 'fs';
+
+import _ from 'lodash';
+import yaml from 'js-yaml';
+
+import { init } from './init';
+import {
+  getResourceSchema,
+  getResultSchema,
+  getSetResultSchema,
+  getPostBodySchema,
+  getPatchBodySchema,
+} from './schemas';
+
+/**
+ * Gets the prefix of the resource schema in components/schemas
+ *
+ * @param resourceName
+ * @returns The prefix
+ */
+const getResourceSchemaPrefix = (resourceName: string) => _.capitalize(resourceName);
+
+/**
+ * Gets the method for a given endpoint. Example: `getById` -> `get`
+ *
+ * @param endpoint
+ * @returns The method
+ */
+const getEndpointMethod = (endpoint: string): string => endpoint.match(/^[a-z]+/)![0];
```

Comment:
Should we also check if the matched string is either "get", "post", "patch" or "delete"?

Reply:
Looks like this is handled else where.

Reply:
This seems like something that should be enforced by the function signature if possible. Possibly using a string enum? https://www.typescriptlang.org/docs/handbook/enums.html#string-enums